### PR TITLE
Add script to also wrap unzip, gunzip and cat steps

### PR DESCRIPTION
**Pull request by samjewell**
Part 2 (of 3/4) of Pivotal ticket: https://www.pivotaltracker.com/story/show/159329673

---

The event data on our Amazon S3 bucket is organised into
one zip file per day of event data. For a given single
zip file containing a single day of data we need to:

1. unzip the file
2. gunzip all of the files that result from unzipping
3. cat all those event files together into a single jsonl stream
4. pipe that stream of data into the `all_processing_steps` script

Previously we would zip the final, processed jsonl file, as the
Python script that pushes events into the Amplitude API expected
a zip file. But there's no easy way to pipe a unix stream into a zip
file, so I instead updated the Python script to work with normal
(not zipped) input jsonl files.

### CODE REVIEW

**Comment by samjewell** [comment]
I have run this script and then run the python uploader on the output (ie. followed the newly updated steps in the Readme), and found it to be good.

See the data present in an Amplitude test project here (reviewer please ask me to give you access to this project): https://analytics.amplitude.com/samtestimport/settings/project/208535/events

File: sample_import.py [hunk 1]
```diff
@@ -118,8 +118,7 @@ def main():
 
 	rownum = 0
 	cur_events = []
-	zf = ZipFile(filename, 'r', allowZip64=True)
-	with zf.open(zf.infolist()[0]) as f:
+	with open(filename) as f:
```
Comment (samjewell) [inline sample_import.py]: I wasn't sure whether I should add `buffering=1` or even `buffering=n>1` here. I only have one example input file so far, which is 28412 lines of jsonl.

Comment (davidmillican) [reply sample_import.py]: I'd imagine that this is ok, but if we hit performance issues you could try running it with a line buffer (value 1). 

Disclaimer: I have never really used this before, so just tried it now. It's possible it'll make a bigger difference than I'm imagining for large files.

File: README.md [hunk 2]
```diff
@@ -9,15 +9,10 @@ Run `make` at the top level of the `jq_scripts` repo to run tests on the jq norm
...
-    `zip processed_events.jsonl.zip processed_events.jsonl`
-7. Run the `sample_import.py` import script over the events file
+1. Get a folder of BridgeU Segment.io data from our S3 bucket, of the format `YYYY-MM-DD.zip`
+1. Processing that day's worth of data:
+    `jq_scripts/exe/e2e_full_day_of_events $PATH_TO/YYYY-MM-DD.zip`
+1. Run the `sample_import.py` import script over the events file
    `source activate python_2`
    `pip install futures`
-   `python sample_import.py processed_events.jsonl.zip 0 --pdb`
+   `python sample_import.py $PATH_TO/YYYY-MM-DD-processed_events.jsonl 0 --pdb`
```
Comment (davidmillican) [inline README.md]: Maybe worth adding a note that the `--pdb` flag will drop into the debugger if it hits an error?

File: exe/e2e_full_day_of_events [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/zsh
+
+date=$(echo $1 | sed 's/.zip//')
```
Comment (davidmillican) [inline exe/e2e_full_day_of_events]: Can we call this `file_date` or something instead. `date` is a bash function so might be a bit confusing.

**Pull request merged**